news/_posts/2020-11-01-prestashop-in-2019-and-beyond-part-4-connecting-the-dots.md

```diff
@@ -0,0 +1,182 @@
+
+This is why we think that the sensible thing to do is to split the work in chunks, put them in a neat roadmap, work on one subset at a time, and deliver them progressively. Of course, it will probably require more effort during a longer timeframe than doing it in one shot. But on the other hand, **performing progressive and limited changes in well-defined parts of the system and delivering them iteratively in predictable milestones** will allow the community to keep up with the Core, as the changes will be smaller and less expensive to adopt than if the whole thing was done at once.
+
+Therefore, we think the best way to advance towards the Future Architecture is through planned, more frequent major releases that focus on refactoring, followed by one or two minor releases focusing on new features. The exact timeline, however, is yet to be determined.
+
+Now, let's see how we are planning to move forward.
+
+## Reduce entropy to increase reliability
+
+A stable system relies on pre-established business rules being respected at all times. If a system's business logic requires all products to have a name and a non-negative price, then it is _expected_ that if a product exists, it will adhere to these rules. Unfulfilled expectations produce inconsistencies, which result in errors.
+
+PrestaShop is a very complex system. As such, it needs to enforce a very high number of business rules corresponding to a wide range of use cases. One of the biggest problems of the current architecture is that extensions can tap anywhere into the Core: since the system is very complex, extensions hooking, overriding or just _using_ Core classes can lead to problems if some business rule is overlooked by a developer who was most likely not even aware of the existence of such rule.
+
+For example, when an order's carrier is updated, an email should be sent to the customer informing them that the order is on its way. This is done automatically when the action is performed using the appropriate service. However, if the order's carrier is modified manually through ObjectModel, for example by a module developer not aware of this rule, then the required email is not sent.
+
+This problem of thick service layers manipulating thin models devoid of domain logic was identified over 15 years ago by Martin Fowler, who described it the [Anemic Domain Model][anemic-domain-model] anti-pattern. His answer was simple: domain logic should be put in the entities.
```

Comment:
Maybe explain what is "domain" and what is "entity" or provide references to understand the concepts behind, in order to avoid confusion with existing concepts inside PrestaShop

Reply:
I spoke about domain in [part 3](https://build.prestashop.com/news/prestashop-in-2019-and-beyond-part-3-the-future-architecture/), but I'll add a link to entity.

Reply:
Because the timeframe between parts 3 and 4 has been quite long, the people who read part 3 may have forgotten. Maybe a simple reminder "domain has been introduced in part 3, you can read it again if the notion is not familiar for you" would be enough?